twill-core/src/main/java/org/apache/twill/internal/TwillRuntimeSpecification.java

```diff
@@ -41,11 +41,23 @@
   private final String rmSchedulerAddr;
   private final Map<String, Map<String, String>> logLevels;
   private final Map<String, Integer> maxRetries;
+  private Double heapToReserveRatio;
```

Comment:
Shouldn't allow null. This is an internal class, you can simply require the caller to pass in valid value.